edgecore panic when there is no default route

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
**What happened**:
When launching edgecore , I get the following crash:

I0213 15:39:31.399239   14361 eventbus.go:61] Init Sub And Pub Client for externel mqtt broker tcp://127.0.0.1:1883 successfully
I0213 15:39:31.399325   14361 client.go:86] edge-hub-cli subscribe topic to $hw/events/device/+/state/update
I0213 15:39:31.399752   14361 client.go:86] edge-hub-cli subscribe topic to $hw/events/device/+/twin/+
I0213 15:39:31.400211   14361 client.go:86] edge-hub-cli subscribe topic to $hw/events/node/+/membership/get
I0213 15:39:31.400477   14361 client.go:86] edge-hub-cli subscribe topic to SYS/dis/upload_records
I0213 15:39:31.407805   14361 proxy.go:92] [L4 Proxy] proxy is running now
I0213 15:39:31.426699   14361 cpu_manager.go:173] [cpumanager] starting with none policy
I0213 15:39:31.426731   14361 cpu_manager.go:174] [cpumanager] reconciling every 0s
I0213 15:39:31.426758   14361 policy_none.go:43] [cpumanager] none policy: Start
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2506348]

goroutine 102 [running]:
k8s.io/kubernetes/pkg/kubelet/cm.(*containerManagerImpl).enforceNodeAllocatableCgroups(0xc0000dab40, 0x6, 0x0)
        /root/.gvm/pkgsets/go1.13.4/global/pkg/mod/k8s.io/kubernetes@v1.17.1/pkg/kubelet/cm/node_container_manager_linux.go:78 +0x1e8
k8s.io/kubernetes/pkg/kubelet/cm.(*containerManagerImpl).setupNode(0xc0000dab40, 0xc0004bc670, 0x0, 0x11)
        /root/.gvm/pkgsets/go1.13.4/global/pkg/mod/k8s.io/kubernetes@v1.17.1/pkg/kubelet/cm/container_manager_linux.go:452 +0xfb
k8s.io/kubernetes/pkg/kubelet/cm.(*containerManagerImpl).Start(0xc0000dab40, 0x0, 0xc0004bc670, 0x33a64a0, 0x48272a8, 0x7f3c8cbf9b00, 0xc000336a00, 0x34535c0, 0xc00062e780, 0xc0001efda0, ...)
        /root/.gvm/pkgsets/go1.13.4/global/pkg/mod/k8s.io/kubernetes@v1.17.1/pkg/kubelet/cm/container_manager_linux.go:600 +0x10a
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).initializeModules(0xc0000eeb00, 0xc000336a00, 0xc00051b7a0)
        /root/codes/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:564 +0x13e
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).Start(0xc0000eeb00)
        /root/codes/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:264 +0x282
created by github.com/kubeedge/beehive/pkg/core.StartModules
        /root/codes/src/github.com/kubeedge/kubeedge/staging/src/github.com/kubeedge/beehive/pkg/core/core.go:23 +0x154


If a connect another ethernet cable that provides a DHCP address and a default route, edgecore is launched successfully

  
**What you expected to happen**:
Sucessful launch of edgecore even if there is no default route.
Default route seems to be needed to get IP address of edge (could be obtained from edge.yaml file ?)

**How to reproduce it (as minimally and precisely as possible)**:
Configure only one ethernet interface with a fixed IP address for a local network (192.168...)
Launch edgecore

**Anything else we need to know?**:
Issue is coming from error returned in kubeedge/edge/pkg/edged/edged.go:
func (e *edged) initializeModules() error {
	node, _ := e.initialNode()
	if err := e.containerManager.Start(node, e.GetActivePods, edgedutil.NewSourcesReady(), e.statusManager, e.runtimeService); err != nil {
		klog.Errorf("Failed to start device plugin manager %v", err)
		return err
	}
	return nil
}
In case initialNode returns an error, it is not handled. In my case the error was raised by util.GetLocalIP because threre is no default route on my edge.
Could it be possible to use the node-ip provided in the edge.yaml file for that purpose ?

**Environment**:
- KubeEdge version(e.g. `cloudcore/edgecore --version`):  
KubeEdge v1.2.0

**CloudSide Environment**:
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

**EdgeSide Environment**:
- edgecore version (e.g. `edgecore --version`):
KubeEdge v1.2.0

- Hardware configuration (e.g. `lscpu`):
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               92
Model name:          Intel(R) Atom(TM) Processor E3930 @ 1.30GHz
Stepping:            9
CPU MHz:             1190.443
CPU max MHz:         1800.0000
CPU min MHz:         800.0000
BogoMIPS:            2572.80
Virtualization:      VT-x
L1d cache:           24K
L1i cache:           32K
L2 cache:            1024K
NUMA node0 CPU(s):   0,1
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg cx16 xtpr pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave rdrand lahf_lm 3dnowprefetch cpuid_fault cat_l2 ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust smep erms mpx rdt_a rdseed smap clflushopt intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts md_clear arch_capabilities

- OS (e.g. `cat /etc/os-release`):
NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"

- Kernel (e.g. `uname -a`):
Linux ... 3 4.15.0-76-generic #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- Go version (e.g. `go version`): 
go version go1.13.4 linux/amd64

- Others:

--- a/edge/pkg/edged/edged.go
+++ b/edge/pkg/edged/edged.go
@@ -560,8 +560,14 @@ func newEdged(enable bool) (*edged, error) {
 }
 
 func (e *edged) initializeModules() error {
-	node, _ := e.initialNode()
-	if err := e.containerManager.Start(node, e.GetActivePods, edgedutil.NewSourcesReady(), e.statusManager, e.runtimeService); err != nil {
+	node, err := e.initialNode()
+	if err != nil {
+		klog.Errorf("Failed to initialNode %v", err)
+		return err
+	}
+
+	err = e.containerManager.Start(node, e.GetActivePods, edgedutil.NewSourcesReady(), e.statusManager, e.runtimeService)
+	if err != nil {
 		klog.Errorf("Failed to start device plugin manager %v", err)
 		return err
 	}

--- a/edge/pkg/edged/edged_status.go
+++ b/edge/pkg/edged/edged_status.go
@@ -299,6 +299,9 @@ func (e *edged) setGPUInfo(nodeStatus *edgeapi.NodeStatusRequest) error {
 }
 
 func (e *edged) getIP() (string, error) {
+	if nodeIP := config.Get().NodeIP; nodeIP != "" {
+		return nodeIP, nil
+	}
 	hostName, _ := os.Hostname()
 	if hostName == "" {
 		hostName = e.nodeName

